php/class-customize-snapshot-manager.php

```diff
@@ -381,6 +389,235 @@ function suspend_kses_for_snapshot_revision_restore() {
 	}
 
 	/**
+	 * Filter for displaying the Snapshot menu location values.
+	 *
+	 * @param array $menu_locations Default menu locations.
+	 * @return array Modified menu locations.
+	 */
+	public function filter_theme_mod_nav_menu_locations( $menu_locations ) {
+		if ( false === $this->snapshot->is_preview() ) {
+			return $menu_locations;
+		}
+
+		$values = $this->snapshot->values();
+		$locations = get_registered_nav_menus();
+
+		foreach ( $locations as $location => $name ) {
+			if ( isset( $values[ 'nav_menu_locations[' . $location . ']' ] ) ) {
+				$menu_locations[ $location ] = $values[ 'nav_menu_locations[' . $location . ']' ];
+			}
+		}
+
+		return $menu_locations;
+	}
+
+	/**
+	 * Filter wp_get_nav_menus() to load Snapshot values.
+	 *
+	 * @see wp_get_nav_menus()
+	 *
+	 * @param array $menus Array of menus.
+	 * @return array Modified array of menus.
+	 */
+	public function filter_wp_get_nav_menus( $menus ) {
+		if ( false === $this->snapshot->is_preview() ) {
+			return $menus;
+		}
+
+		$values = $this->snapshot->values();
+		$removed = array();
+
+		foreach ( $values as $setting_id => $value ) {
+			if ( preg_match( '/^nav_menu\[(?P<id>-?\d+)\]$/', $setting_id, $matches ) ) {
+				if ( false !== $value ) {
+					$menus[] = $this->get_nav_menu_object( $matches['id'], $value );
+				} else {
+					$removed[] = intval( $matches['id'] );
+				}
+			}
+		}
+
+		if ( ! empty( $removed ) && ! empty( $menus ) ) {
+			foreach ( $menus as $key => $term ) {
+				if ( in_array( $term->term_id, $removed, true ) ) {
+					unset( $menus[ $key ] );
+				}
+			}
+		}
+
+		return array_values( $menus );
+	}
+
+	/**
+	 * Filter wp_get_nav_menu_items() to load Snapshot values.
+	 *
+	 * @see wp_get_nav_menu_items()
+	 *
+	 * @param array  $items An array of menu item post objects.
+	 * @param object $menu  The menu object.
+	 * @param array  $args  An array of arguments used to retrieve menu item objects.
+	 * @return array Array of menu items.
+	 */
+	function filter_wp_get_nav_menu_items( $items, $menu, $args ) {
```

Comment:
I wonder why this is needed as opposed to `WP_Customize_Nav_Menu_Item::preview()` which adds its own filter handler of the same name? https://github.com/xwp/wordpress-develop/blob/266ffe1f35e577954817d453cd7b2d84778cb57c/src/wp-includes/customize/class-wp-customize-nav-menu-item-setting.php#L368-L480